### PR TITLE
Angus/build updates

### LINKED ISSUE #321: migrate create-react-app-typescript to Create React App
ref: https://vincenttunru.com/migrate-create-react-app-typescript-to-create-react-app/

### LINKED ISSUE #706: Migrate to react-scripts instead of react-scripts-ts
See discussion in PR #683 and PR #527

### PR DESCRIPTION
**Pull request by veggiesaurus**
Closes #321 and closes #706 

This is a rather large change to the build, because it now uses `create-react-app@3.4.3`, which comes with new build scripts. Some other changes:

- allows us to import SCSS files directly, no need to run all the CSS-watching stuff
- uses eslint (with typescript plugins) instead of tslint
- uses browserlist to ensure that we compile to a sensible target, rather than supporting ancient browsers

There are a few stricter eslint rules by default, and this has required a number of minor changes to the code. Hopefully this doesn't break anything. 

@kswang1029 can you do some testing of the frontend in general?
@ajm-asiaa can you try build electron apps using this branch?
@duidae @panchyo0 @TienHao can you check through the changes to the code itself? Most were suggested directly by eslint.

note: this seems to break the breadcrumbs fix by @ajm-asiaa, I'll look into solving this during the review process

### CODE REVIEW

**Comment by ajm-ska** [comment]
@veggiesaurus Does the result of `npm run build` work for you? I only seem to get a blank screen. The debug console shows the following:
Screenshot 2020-10-07 at 20 29 43

**Comment by veggiesaurus** [comment]
> @veggiesaurus Does the result of `npm run build` work for you? I only seem to get a blank screen. The debug console shows the following:
> Screenshot 2020-10-07 at 20 29 43
> 

Perhaps you need to force a hard reload?

**Comment by ajm-ska** [comment]
> > @veggiesaurus Does the result of `npm run build` work for you? I only seem to get a blank screen. The debug console shows the following:
> > Screenshot 2020-10-07 at 20 29 43
> 
> Perhaps you need to force a hard reload?

That doesn't seem to work (holding shift and hitting reload). 
I see the automatic builds on the [Mix 'N Match](http://carta.asiaa.sinica.edu.tw/builds/) page have the same issue.

**Comment by veggiesaurus** [comment]
> > > @veggiesaurus Does the result of `npm run build` work for you? I only seem to get a blank screen. The debug console shows the following:
> > > Screenshot 2020-10-07 at 20 29 43
> > 
> > 
> > Perhaps you need to force a hard reload?
> 
> That doesn't seem to work (holding shift and hitting reload).
> I see the automatic builds on the [Mix 'N Match](http://carta.asiaa.sinica.edu.tw/builds/) page have the same issue.

It might be because I removed the `homepage` setting in `package.json`. I've added it back, can you see if that solves it?

**Comment by ajm-ska** [comment]
> > > > @veggiesaurus Does the result of `npm run build` work for you? I only seem to get a blank screen. The debug console shows the following:
> > > > Screenshot 2020-10-07 at 20 29 43
> > > 
> > > 
> > > Perhaps you need to force a hard reload?
> > 
> > 
> > That doesn't seem to work (holding shift and hitting reload).
> > I see the automatic builds on the [Mix 'N Match](http://carta.asiaa.sinica.edu.tw/builds/) page have the same issue.
> 
> It might be because I removed the `homepage` setting in `package.json`. I've added it back, can you see if that solves it?

Yes. That fixed the problem! It works now. It appears to work when put into Electron too.

**Review by ajm-ska — APPROVED** [review]
The building aspects seem to be fine. The production build works now and the Electron app version can still be produced. Using the Docker build process seems to work fine too.

**Comment by kswang1029** [comment]
I am trying to explore the GUI from time to time to see if there are odd things and report here.
PART1
1. the checkbox of the molecular line transition table in the spectral line query widget and preference-log event is broken.
Screen Shot 2020-10-08 at 8 48 49 AM
Screen Shot 2020-10-08 at 9 17 25 AM

2. breadcrumbs in the file browser is broken
Screen Shot 2020-10-08 at 9 03 40 AM

File: src/components/Shared/ProfilerInfo/ProfilerInfoComponent.tsx [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
 interface ProfilerInfoComponentProps {
     info: string[];
 }
 
 export const ProfilerInfoComponent: React.FC<ProfilerInfoComponentProps> = (props) => {
-    let infoString = "";
-    props.info?.forEach((info, index) => {
-        infoString = index === 0 ? info : (infoString + ", " + info);
-    });
+    let infoTags = null;
```
Comment (duidae) [inline src/components/Shared/ProfilerInfo/ProfilerInfoComponent.tsx]: seems previous PR #1216  is not merged

File: src/stores/FrameStore.ts [hunk 2]
```diff
@@ -1072,16 +1073,10 @@ export class FrameStore {
             }
         }
 
-        const imageX = Math.round(cursorPosImageSpace.x);
```
Comment (duidae) [inline src/stores/FrameStore.ts]: PR #1221 is also missing

**Review by duidae — CHANGES_REQUESTED** [review]
PR #1216 & #1221 are missing

File: package.json [hunk 3]
```diff
@@ -40,22 +42,23 @@
...
+        "react-dom": "^16.13.1",
         "react-flexbox-grid": "^2.1.2",
         "react-google-login": "^5.1.20",
         "react-iframe": "^1.8.0",
         "react-konva": "^16.13.0-3",
         "react-plotly.js": "^2.4.0",
         "react-resize-detector": "^4.2.1",
         "react-rnd": "^9.2.0",
+        "react-scripts": "3.4.3",
         "react-scripts-ts": "^2.17.0",
```
Comment (panchyo0) [inline package.json]: should we remove "react-scripts-ts" from package.json?

**Review by panchyo0 — APPROVED** [review]

**Comment by veggiesaurus** [comment]
The breadcrumbs and checkbox issues have been fixed now. And we no longer need the `crumbs_fix` workaround :+1: 

@kswang1029 can you give it another try when you get the chance?

**Comment by kswang1029** [comment]
@veggiesaurus confirmed with mix n match: both breadcrumbs and checkbox work properly.

**Review by duidae — APPROVED** [review]

**Review by TienHao — APPROVED** [review]

**Pull request merged**